File: web/src/api/miser/miser_transaction.js
import service from '@/utils/request'

export const createMiserTransaction = (data) => {
  return service({
    url: '/miserTransaction/createMiserTransaction',
    method: 'post',
    data
  })
}

export const deleteMiserTransaction = (params) => {
  return service({
    url: '/miserTransaction/deleteMiserTransaction',
    method: 'delete',
    params
  })
}

export const deleteMiserTransactionByIds = (params) => {
  return service({
    url: '/miserTransaction/deleteMiserTransactionByIds',
    method: 'delete',
    params
  })
}

export const updateMiserTransaction = (data) => {
  return service({
    url: '/miserTransaction/updateMiserTransaction',
    method: 'put',
    data
  })
}

export const findMiserTransaction = (params) => {
  return service({
    url: '/miserTransaction/findMiserTransaction',
    method: 'get',
    params
  })
}

export const getMiserTransactionList = (params) => {
  return service({
    url: '/miserTransaction/getMiserTransactionList',
    method: 'get',
    params
  })
}

File: server/router/miser/miser_transaction.go
package miser

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MiserTransactionRouter struct{}

func (s *MiserTransactionRouter) InitMiserTransactionRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	miserTransactionRouter := Router.Group("miserTransaction").Use(middleware.OperationRecord())
	{
		miserTransactionRouter.POST("createMiserTransaction", miserTransactionApi.CreateMiserTransaction)
		miserTransactionRouter.DELETE("deleteMiserTransaction", miserTransactionApi.DeleteMiserTransaction)
		miserTransactionRouter.DELETE("deleteMiserTransactionByIds", miserTransactionApi.DeleteMiserTransactionByIds)
		miserTransactionRouter.PUT("updateMiserTransaction", miserTransactionApi.UpdateMiserTransaction)
	}

	miserTransactionRouterWithoutRecord := Router.Group("miserTransaction")
	{
		miserTransactionRouterWithoutRecord.GET("findMiserTransaction", miserTransactionApi.FindMiserTransaction)
		miserTransactionRouterWithoutRecord.GET("getMiserTransactionList", miserTransactionApi.GetMiserTransactionList)
	}

	miserTransactionRouterWithoutAuth := PublicRouter.Group("miserTransaction")
	{
		_ = miserTransactionRouterWithoutAuth
	}
}
